query-engine/connector-test-kit-rs/query-tests-setup/src/lib.rs

```diff
@@ -292,20 +339,49 @@ fn run_connector_test_impl(
         .unwrap();
         let schema_id = runner.schema_id();
 
-        if let Err(err) = test_fn(runner)
+        let expected_to_fail = ignore_lists::is_expected_to_fail(&full_test_name);
+
+        let test_future = if expected_to_fail {
+            Either::Left(async {
+                match AssertUnwindSafe(test_fn(runner)).catch_unwind().await {
+                    Ok(Ok(_)) => panic!("expected this test to fail but it succeeded"),
+                    Ok(Err(err)) => {
+                        eprintln!("test failed as expected: {err}");
+                        Ok(())
+                    }
+                    Err(_) => {
+                        eprintln!("test panicked as expected");
+                        Ok(())
+                    }
+                }
+            })
+        } else {
+            Either::Right(test_fn(runner))
+        };
+
+        if let Err(err) = test_future
             .with_subscriber(test_tracing_subscriber(ENV_LOG_LEVEL.to_string(), log_tx))
             .with_recorder(recorder)
             .await
         {
             panic!("💥 Test failed due to an error: {err:?}");
         }
 
-        crate::teardown_project(&datamodel, db_schemas, schema_id)
-            .await
-            .unwrap();
+        if let Err(e) = crate::teardown_project(&datamodel, db_schemas, schema_id).await {
+            if !expected_to_fail {
+                panic!("Teardown failed: {e}");
+            }
+        }
     });
 }
 
+fn build_full_test_name(test_fn_full_name: &'static str, original_test_function_name: &'static str) -> String {
+    let mut parts = test_fn_full_name.split("::").skip(1).collect::<Vec<_>>();
+    let last = parts.len() - 1;
+    parts[last] = original_test_function_name;
```

Comment:
```suggestion
    parts.pop();
    parts.push(original_test_function_name);
```
I like to avoid index arithmetic when possible